src/shared/json.c

```diff
@@ -3877,6 +3878,56 @@ int json_buildv(JsonVariant **ret, va_list ap) {
                         break;
                 }
 
+                case _JSON_BUILD_STRING_SET: {
+                        Set *set;
+
+                        if (!IN_SET(current->expect, EXPECT_TOPLEVEL, EXPECT_OBJECT_VALUE, EXPECT_ARRAY_ELEMENT)) {
+                                r = -EINVAL;
+                                goto finish;
+                        }
+
+                        set = va_arg(ap, Set*);
+
+                        if (current->n_suppress == 0) {
+                                _cleanup_free_ void **sorted = NULL;
+                                JsonVariant **array = NULL;
+                                size_t n_sorted;
+
+                                CLEANUP_ARRAY(array, n_sorted, json_variant_unref_many);
+
+                                r = set_dump_sorted(set, &sorted, &n_sorted);
```

Comment:
maybe we should teach set_dump_sorted() the trick to implicitly append a zero'ed out entry to the end of the array it returns (which is not counted).

That way, the `sorted` array would be one of those regular string arrays you can then turn into a json variant with a simple json_variant_new_strv()